From fae226d72327c3c5594c53138978fb2eeca0402b Mon Sep 17 00:00:00 2001
Message-Id: <fae226d72327c3c5594c53138978fb2eeca0402b.1425610844.git.chang-joon.lee@intel.com>
In-Reply-To: <6fa2ab23adf6d7de781e906d04a0224410262d9b.1425610844.git.chang-joon.lee@intel.com>
References: <6fa2ab23adf6d7de781e906d04a0224410262d9b.1425610844.git.chang-joon.lee@intel.com>
From: "yifei.xue" <yifei.xue@intel.com>
Date: Thu, 29 Jan 2015 14:47:27 +0800
Subject: [PATCH 24/27] SQUASHME! [VPG]: drm/i915: return correct backlight
 value for LPIO PWM

The LPIO PWM hardware keeps the signal high in duty cycle for a duration
of 0xFF - brightness level. Hence when reading the brightness value from
hardware, the brightness level must be inverted.

In current implementation, the brightness level read from LPIO PWM
hardware is not inverted resulting in incorrect brightness value
returned. This patch fixes this issue.

SQUASHME!: This patch should be squashed into following patch:
     Author: Deepak M <m.deepak@intel.com>
     FOR_UPSTREAM [VPG]: drm/i915: Added the backlight support for the MIPI panel
     Change-Id: I45e7288d38833b487e179920275dfe091f6d6d66

Issue: IMINAN-19533
Change-Id: I5a033e6650b0fcfb41049c4d104ea1b99f444c17
Signed-off-by: yifei.xue <yifei.xue@intel.com>
---
 drivers/gpu/drm/i915/intel_panel.c |    2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/drivers/gpu/drm/i915/intel_panel.c b/drivers/gpu/drm/i915/intel_panel.c
index 0e92f9e..6b6b746 100644
--- a/drivers/gpu/drm/i915/intel_panel.c
+++ b/drivers/gpu/drm/i915/intel_panel.c
@@ -503,7 +503,7 @@ static u32 vlv_get_mipi_backlight(struct intel_connector *connector)
 	struct drm_i915_private *dev_priv = dev->dev_private;
 
 	if (dev_priv->vbt.dsi.config->pmic_soc_blc)
-		return lpio_bl_read(0, LPIO_PWM_CTRL) & 0xff;
+		return (~lpio_bl_read(0, LPIO_PWM_CTRL)) & 0xff;
 	else
 		return intel_soc_pmic_readb(PMIC_PWM_LEVEL);
 }
-- 
1.7.9.5

